parser.py

```diff
@@ -1,3 +1,4 @@
+from abc import abstractproperty
```

Comment:
what's this for?

Reply:
I don't know how this got here really :). Removed this now